doc/source/groupby.rst

```diff
@@ -521,9 +521,17 @@ Transformation
 --------------
 
 The ``transform`` method returns an object that is indexed the same (same size)
-as the one being grouped. Thus, the passed transform function should return a
-result that is the same size as the group chunk. For example, suppose we wished
-to standardize the data within each group:
+as the one being grouped. The transform function must:
+
+* Return a result that is either the same size as the group chunk or
+  broadcastable to the size of the group chunk.
+* Operate column-by-column on the group chunk.  A fast path is used if the
+  transform function also operates on the entire group chunk as a DataFrame.
```

Comment:
I think this will not be very clear to readers. Do you have a specific example where you noticed this? Or is this explanation based on looking at the code?